synapse/handlers/directory.py

```diff
@@ -204,6 +204,10 @@ async def delete_association(
             )
 
         room_id = await self._delete_association(room_alias)
+        # The call to `_user_can_delete_alias` above ensures the alias exists.
+        # `_delete_association` returns `None` only if the alias doesn't exist.
+        # We leave the assert here for mypy's benefit.
+        assert room_id is not None
```

Comment:
Isn't this actually hiding a race condition?

Reply:
Thanks, you're right. Maybe something deletes the alias after we get `can_delete: True` but before `_delete_association` finishes.